### PR TITLE
CASMINST-6896 Add Update NCN images step

### PR DESCRIPTION
**Pull request by rustydb**
# Description

This includes a new step in the 1.5.2 upgrade that rebuilds NCN images, pulling in any changes from csm-config.

Namely, this ensures that we build new images that include the new DST GPG key and that these new images are readily available for an administrator if/when they must add or rebuild an NCN.

# Checklist

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: upgrade/1.5.2/CSM-Only.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+   true
+   ```
+
+1. (`ncn-m001#`) Set environment variables with the IMS image IDs.
+
+    * Kubernetes image IMS ID
+
+      ```bash
+      NEW_KUBERNETES_IMAGE_ID="$(cray cfs sessions describe "${K8S_CFS_SESSION_NAME}" --format json | jq -r '.status.artifacts[].result_id')"
+      export NEW_KUBERNETES_IMAGE_ID
```
Comment (mharding-hpe) [inline upgrade/1.5.2/CSM-Only.md]: Do we need to export this? Looks like we're passing it into the script as an argument.

```suggestion
```

File: upgrade/1.5.2/CSM-Only.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      echo "NEW_KUBERNETES_IMAGE_ID=$NEW_KUBERNETES_IMAGE_ID"
+      ```
+
+    * Storage image IMS ID
+
+      ```bash
+      NEW_STORAGE_IMAGE_ID="$(cray cfs sessions describe "${CEPH_CFS_SESSION_NAME}" --format json | jq -r '.status.artifacts[].result_id')"
+      export NEW_STORAGE_IMAGE_ID
```
Comment (mharding-hpe) [inline upgrade/1.5.2/CSM-Only.md]: Do we need to export this? Looks like we're passing it into the script as an argument.

```suggestion
```

File: upgrade/1.5.2/CSM-Only.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      echo "KUBERNETES_IMAGE_ID=$KUBERNETES_IMAGE_ID"
+      ```
+
+    * Storage-CEPH ID
+
+      ```bash
+      STORAGE_IMAGE_ID="$(kubectl -n services get cm cray-product-catalog -o jsonpath='{.data.csm}' \
+        | yq r -j - '"'${CSM_RELEASE}'".images' \
+        | jq -r '. as $o | keys_unsorted[] | select(startswith("secure-storage")) | $o[.].id')"
+      export STORAGE_IMAGE_ID
```
Comment (mharding-hpe) [inline upgrade/1.5.2/CSM-Only.md]: Do we need to export this? Looks like we're just using it as a command argument later.

```suggestion
```

File: upgrade/1.5.2/CSM-Only.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+1. (`ncn-m001#`) Set the [IMS](../../glossary.md#image-management-service-ims) IDs for each Management node sub-role.
+
+    * Kubernetes ID
+
+      ```bash
+      KUBERNETES_IMAGE_ID="$(kubectl -n services get cm cray-product-catalog -o jsonpath='{.data.csm}' \
+        | yq r -j - '"'${CSM_RELEASE}'".images' \
+        | jq -r '. as $o | keys_unsorted[] | select(startswith("secure-kubernetes")) | $o[.].id')"
+      export KUBERNETES_IMAGE_ID
```
Comment (mharding-hpe) [inline upgrade/1.5.2/CSM-Only.md]: Do we need to export this? Looks like we're just using it as a command argument later.

```suggestion
```

**Review by mharding-hpe — APPROVED** [review]

**Review by jpdavis-prof — APPROVED** [review]

**Comment by rustydb** [comment]
Spellcheck is failing on the change I made for passing the stylecheck. I am going to have to bypass one or the other.

**Pull request merged**